tracer/src/Datadog.Trace/Agent/StatsAggregator.cs

```diff
@@ -0,0 +1,197 @@
+// <copyright file="StatsAggregator.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Threading.Tasks;
+using Datadog.Trace.Configuration;
+using Datadog.Trace.ExtensionMethods;
+using Datadog.Trace.Logging;
+using Datadog.Trace.PlatformHelpers;
+
+namespace Datadog.Trace.Agent
+{
+    internal class StatsAggregator : IStatsAggregator
+    {
+        private const int BufferCount = 2;
+
+        private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor<StatsAggregator>();
+
+        private readonly StatsBuffer[] _buffers;
+
+        private readonly IApi _api;
+
+        private readonly TaskCompletionSource<bool> _processExit;
+
+        private readonly TimeSpan _duration;
```

Comment:
What is this `_duration` used for? Looks like a flush timeout. Can you rename it or add a comment? Thanks.

Reply:
In 4b4f532d43f18b26f2fbb065d6f6e130cea2cf4c I modified the variable name "duration" to "bucketDuration" so it better indicates that this is the length of the window used for grouping stats. Let me know if you have any other suggestions